testing/tests/DevExpress.ui.widgets/treeViewParts/expandedItems.js

```diff
@@ -690,4 +692,24 @@ module("Expanded items", {
 
         assert.equal(contentReadyStub.callCount, 2, "event is thrown twice");
     });
+
+    ['dataSource', 'items'].forEach((optionName) => {
+        QUnit.test(`Nodes expanding works even with loop/cycle in ${optionName} option`, function(assert) {
+            let options = { dataStructure: "plain", rootValue: 1 };
+            options[optionName] = [
+                { id: 1, text: "item1", parentId: 2, selected: false, expanded: false },
+                { id: 2, text: "item1_1", parentId: 1, selected: false, expanded: false }];
+
+            let treeView = createInstance(options),
+                $parentItem = treeView.getElement().find('[aria-level="1"]'),
+                $childItem = treeView.getElement().find('[aria-level="2"]');
+
+            assert.equal($childItem.length, 0);
+
+            treeView.instance.expandItem($parentItem);
+            $childItem = treeView.getElement().find('[aria-level="2"]');
+            assert.equal($childItem.length, 1);
+            assert.equal(treeView.hasInvisibleClass($childItem), false);
```

Comment:
Do we need to check hasInvisibleClass value before 'expandItem'?

Reply:
No, the node is not rendered, so it doesn't exists int he DOM